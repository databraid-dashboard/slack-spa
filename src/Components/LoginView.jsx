// @flow

import React from 'react';
import PropTypes from 'prop-types';
import { Button, Image } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import logo from '../images/slack-logo.png';
// import slackConnectHref from '../Constants/connectWithSlackHref';
import slack from '../images/slackIcon.png';
import { connectWithSlack } from '../Actions/index';


export function LoginView(props) {
  const { connectWithSlack } = props;

  return (
    <div>
      <div className="SlackApp">
        <div className="SlackApp-header">
          <img src={logo} className="SlackApp-logo" alt="slack-logo" />
        </div>
        <div className="connectButton">
          <Button
            size="big"
            color={'teal'}
            // href={slackConnectHref}
            onClick={
              () => connectWithSlack()
            }
          >
            <Image avatar src={slack} />
            Connect with Slack
          </Button>
        </div>
      </div>
    </div>
  );
}

LoginView.defaultProps = {
  connectWithSlack: PropTypes.function,
};


export const mapDispatchToProps = dispatch =>
  bindActionCreators({ connectWithSlack }, dispatch);

export default connect(null, mapDispatchToProps)(LoginView);
