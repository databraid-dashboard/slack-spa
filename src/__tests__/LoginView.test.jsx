import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import LoginView from '../LoginView';

describe('Renders message with default props', () => {
  it('should render a message when given props', () => {
    const loginView = shallow(<LoginView />);
    expect(toJson(loginView)).toMatchSnapshot();
  });
});
