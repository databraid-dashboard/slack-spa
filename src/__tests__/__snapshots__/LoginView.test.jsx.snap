// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders message with default props should render a message when given props 1`] = `
<div>
  <div
    className="SlackApp"
  >
    <div
      className="SlackApp-header"
    >
      <img
        alt="slack-logo"
        className="SlackApp-logo"
        src="slack-logo.png"
      />
    </div>
    <div
      className="connectButton"
    >
      <Button
        as="button"
        color="teal"
        onClick={[Function]}
        size="big"
      >
        <Image
          as="img"
          avatar={true}
          src="slack_icon.png"
          ui={true}
        />
        Connect with Slack
      </Button>
    </div>
  </div>
</div>
`;
